cranelift/codegen/src/verifier/mod.rs

```diff
@@ -1310,6 +1330,23 @@ impl<'a> Verifier<'a> {
                     .map(|&v| self.func.dfg.value_type(v));
                 self.typecheck_variable_args_iterator(inst, iter, errors)?;
             }
+            BranchInfo::Conditional(block_then, _args_then, block_else, _args_else) => {
+                let iter = self
+                    .func
+                    .dfg
+                    .block_params(block_then)
+                    .iter()
+                    .map(|&v| self.func.dfg.value_type(v));
+                self.typecheck_variable_args_iterator(inst, iter, errors)?;
+
+                let iter = self
+                    .func
+                    .dfg
+                    .block_params(block_else)
+                    .iter()
+                    .map(|&v| self.func.dfg.value_type(v));
+                self.typecheck_variable_args_iterator(inst, iter, errors)?;
+            }
```

Comment:
I think this is the first place that `typecheck_variable_args_iterator` is called twice with the same instruction. Since it compares the given iterator against `inst_variable_args(inst)`, I wouldn't think that would be the right thing to do. Is it?

Reply:
It shouldn't be called at all, thanks for catching this! This was a hold-over from when the block args were held in the same ValueList.

Reply:
Scratch that, it should be called from here: how else would we typecheck the branches of `brif` :)

I think the issue here is that it's assumed that the args always come from the instruction. I'll rework this to take the args as a separate argument, so that it's possible to check the args for a specific block.